internal/ast/ast.go

```diff
@@ -292,12 +292,17 @@ func (e *NumExpr) String() string {
 	}
 }
 
-// StrExpr is a literal string like "foo".
+// StrExpr is a literal string like "foo" or a regexp constant like /foo/.
 type StrExpr struct {
-	Value string
+	Value  string
+	Regexp bool
```

Comment:
Naming nit: can we call this `IsRegexp` for clarity?

Reply:
We could, but that's so not-Go

Reply:
Maybe we should document it better?

Reply:
Looks like your mostly right that it's "not Go", though not entirely, [for example, the `Is*` fields here](https://pkg.go.dev/net#DNSError). However, we have another bool in this package, `IncrExpr.Pre`, that doesn't have the `Is`, so let's go with that.

However, on closer look, can we please call it `Regex bool`? The "regexp" with a 'p' kind of refers to Go's "regexp" package, whereas we I tend to call it "regex" in an AWK context (for example, the `RegExpr.Regex` field itself).

Reply:
Done